lightningd/options.c

```diff
@@ -308,8 +308,8 @@ static char *opt_add_proxy_addr(const char *arg, struct lightningd *ld)
 	if (!parse_wireaddr(arg, ld->proxyaddr, 9050,
 			    ld->use_proxy_always ? &needed_dns : NULL,
 			    NULL)) {
-		return tal_fmt(NULL, "Unable to parse Tor proxy address '%s' %s",
-			       arg, needed_dns ? " (needed dns)" : "");
+		return tal_fmt(NULL, "Unable to parse Tor proxy address '%s'",
+			       arg);
```

Comment:
Maybe better: `(ld->use_proxy_always && needed_dns) ? " (needed dns)" : ""`?

Reply:
In this case, it's clearer to simply initialize it to 'false' I think, since it's ONLY for this msg.